lib/utils/ignore.ts

```diff
@@ -115,6 +115,11 @@ export class FileIgnorer {
 			return true;
 		}
 
+		// Don't ignore Dockerfile or docker-compose.yml
+		if (path.basename(relFile) === 'Dockerfile' || path.basename(relFile) === 'docker-compose.yml') {
+			return true;
```

Comment:
I find it uncomfortable that we hardcode these exceptions (it feels like it is the user's mistake if they list 'Dockerfile' in their `.dockerignore` and/or `.gitignore` files), but there is a precedent (`.balena` just above...), it's probably for the best in practice (given support requests) and it is what previous CLI check-in meetings had decided. So, I agree. :-)

In this case, we should probably also hardcode exceptions for `Dockerfile.template`, and the likes of `Dockerfile.armv7hf` and `Dockerfile.raspberrypi3` (see [documentation](https://www.balena.io/docs/learn/develop/dockerfile/#multiple-dockerfiles)). It think it's enough to add a regex along the lines of `(Dockerfile|Dockerfile\.\S+)`.

Reply:
Yeah, I can't really think of an alternative approach.  I think it stems from the fact that we are using `.dockerignore` in a subtly different way to docker.  (e.g. Adding `Dockerfile` to `.dockerignore` is even suggested in the docker documentation).

Reply:
One small downside is that the monica project included `Dockerfile.dev`, which would now be included in the build.